spec.html

```diff
@@ -70,26 +70,279 @@ <h1>Runtime Semantics: Evaluation</h1>
+              1. Perform ! _envRec_.InitializeBinding(_exportName_, *undefined*).
+              1. Return *undefined*.
+            </emu-alg>
+          </emu-clause>
+
+          <emu-clause id="sec-smr-Evaluate">
+            <h1>Evaluate ( )</h1>
+
+            <p>The Evaluate concrete method of a Synthetic Module Record implements the corresponding Module Record abstract method.</p>
+            <p>It performs the following steps:</p>
+
+            <emu-alg>
+              1. Let _module_ be this Synthetic Module Record.
+              1. Let _moduleCxt_ be a new ECMAScript code execution context.
+              1. Set the Function of _moduleCxt_ to *null*.
+              1. Assert: _module_.[[Realm]] is not *undefined*.
+              1. Set the Realm of _moduleCxt_ to _module_.[[Realm]].
+              1. Set the ScriptOrModule of _moduleCxt_ to _module_.
+              1. Set the VariableEnvironment of _moduleCxt_ to _module_.[[Environment]].
+              1. Set the LexicalEnvironment of _moduleCxt_ to _module_.[[Environment]].
+              1. Suspend the currently running execution context.
+              1. Push _moduleCxt_ on to the execution context stack; _moduleCxt_ is now the running execution context.
+              1. Let _result_ be the result of performing ? _module_.[[EvaluationSteps]](_module_).
+              1. Suspend _moduleCxt_ and remove it from the execution context stack.
+              1. Resume the context that is now on the top of the execution context stack as the running execution context.
+              1. Return Completion(_result_).
+            </emu-alg>
+          </emu-clause>
+        </emu-clause>
+      </emu-clause>
+
+  <emu-clause id="sec-create-default-export-synthetic-module" aoid="CreateDefaultExportSyntheticModule">
+    <h1>CreateDefaultExportSyntheticModule ( _defaultExport_ )</h1>
+    <p>The CreateDefaultExportSyntheticModule abstract operation creates a Synthetic Module Record whose default export is _defaultExport_. It performs the following steps:</p>
+    <emu-alg>
+      1. Return CreateSyntheticModule(&laquo; *"default"* &raquo;, the following steps, the current Realm, _defaultExport_) with the following steps given _module_ as an argument:</p>
+        1. SetSyntheticModuleExport(_module_, *"default"*, _module_.[[HostDefined]]).</li>
+    </emu-alg>
+    <!-- TODO: Could we apply the new spec closure concept here rather than painstakingly passing _defaultExport_ through [[HostDefined]]? -->
+  </emu-clause>
+
+  <emu-clause id="sec-parse-json-module" aoid="ParseJSONModule">
+    <h1>ParseJSONModule ( _source_ )</h1>
+    <p>The abstract operation ParseJSONModule takes a single argument _source_ which is a String representing the contents of a module.</p>
+
+    <emu-alg>
+      1. Let _json_ be ? Call(%JSONParse%, *undefined*, « _source_ »).  <!-- TODO: Refactor JSONParse into an abstract algorithm -->
```

Comment:
Should the JSON be frozen? What if a developer accidentally call the `json.arr.sort()` then the file is corrupted

Reply:
Let's discuss this in a separate issue: https://github.com/tc39/proposal-module-attributes/issues/54